sky/serve/replica_managers.py

```diff
@@ -84,11 +95,12 @@ def launch_cluster(task_yaml_path: str,
         retry_cnt += 1
         try:
             usage_lib.messages.usage.set_internal()
+            # remove retry until up to avoid infinite retry on
+            # zone without capacity.
             sky.launch(task,
                        cluster_name,
                        detach_setup=True,
                        detach_run=True,
-                       retry_until_up=True,
```

Comment:
After removing this, we may get the `sky_launch_status == ProcessStatus.FAILED` for replica being launching in a location that has no capacity. Does that mean the user will see a lot of replicas as FAILED in the `sky serve status` if we cannot get the availability every locations? It can be very confusing. Is it possible to either:
1. (preferred) keep the failover in the same replica entry (row) in the serve status table, i.e. reuse the replica id for the new replica that is launched after the old one that does not have capacity.
2. Add a ReplicaStatus.FAILED_NO_CAPACITY status for the replica failed due to the capacity issue.